### ISSUE TITLE
Too much memory used for the implementation of SmallElements - revise "AperyList"

### ISSUE THREAD

**Issue by pedritomelenas** [gap-packages/numericalsgps#44]
This example, due to @coneill-math, shows the memory usage behavior
```
gap> N := 20000;; s := NumericalSemigroup([N..N+1000]);; Maximum(AperyListOfNumericalSemigroup(s)); time;
Error, reached the pre-set memory limit
(change it with the -o command line option) in
  Add( S, x ); at /home/pedro/Repositorios/numericalsgps//gap/elements.gi:288 called from
SmallElementsOfNumericalSemigroup( sgp
 ) at /home/pedro/Repositorios/numericalsgps//gap/elements.gi:432 called from
GapsOfNumericalSemigroup( S ) at /home/pedro/Repositorios/numericalsgps//gap/basics.gi:755 called from
AperyListOfNumericalSemigroupWRTElement( S, MultiplicityOfNumericalSemigroup( S )
 ) at /home/pedro/Repositorios/numericalsgps//gap/basics.gi:779 called from
<function "unknown">( <arguments> )
 called from read-eval loop at *stdin*:54
you can 'return;'
brk> return;
```

**Comment by pedritomelenas** [comment]
It seems to be a problem with the case of proportionally modular numerical semigroups

**Comment by ManuelAFDelgado** [comment]
The problem was partially solved by adding a method to compute the small elements of a numerical semigroup given by a closed interval (commit: 76d1beb).

Computing Apery sets still needs to be revised:
gap> N := 20000;; s := NumericalSemigroup([N..N+1000]);; Maximum(AperyListOfNumericalSemigroup(s)); time;
419999
400006

**Comment by pedritomelenas** [comment]
gap> N := 20000;; s := NumericalSemigroup([N..N+1000]);; Maximum(AperyListOfNumericalSemigroup(s)); time;
419999
884
